### PR TITLE
[PLAT-197] [PLAT-213] Clean up excess logger usage

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
PR to clean up logger usage, and switch us to using namespaced loggers.

## 👌 Resolves:
- PLAT-197
- PLAT-213

## 📦 Impacts:
everything

## ✅ PR Checklist
- [x] Convert all logs to namespaced logs, clean up excess

### CODE REVIEW

File: packages/mds-agency-cache/index.ts [hunk 1]
```diff
@@ -100,7 +99,7 @@ async function getEventsInBBox(bbox: BoundingBox) {
   const events: string[] = await client.georadius(decorateKey('locations'), lng, lat, radius, 'm')
   const finish: Timestamp = now()
   const timeElapsed = finish - start
-  logger.info(`mds-agency-cache getEventsInBBox ${JSON.stringify(bbox)} time elapsed: ${timeElapsed}ms`)
+  AgencyCacheLogger.debug(`mds-agency-cache getEventsInBBox ${JSON.stringify(bbox)} time elapsed: ${timeElapsed}ms`)
```
Comment (drtyh2o) [inline packages/mds-agency-cache/index.ts:102]: Having `mds-agency-cache` in the log text seems redundant now.

File: packages/mds-agency-cache/index.ts [hunk 2]
```diff
@@ -181,16 +180,15 @@ async function wipeDevice(device_id: UUID) {
     decorateKey(`device:${device_id}:device`)
   ]
   if (keys.length > 0) {
-    logger.info('mds-agency-cache::wipeDevice, deleting keys', { keys })
+    AgencyCacheLogger.debug('mds-agency-cache::wipeDevice, deleting keys', { keys })
```
Comment (drtyh2o) [inline packages/mds-agency-cache/index.ts:183]: Ditto. Should be able to remove all the prefixes from the log text.

File: packages/mds-agency-cache/logger.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import logger from '@mds-core/mds-logger'
+
+export const AgencyCacheLogger = logger.createLogger('AgencyCache')
```
Comment (drtyh2o) [inline packages/mds-agency-cache/logger.ts:3]: Should we use `mds-agency-cache` as the name to make associating the log with the package easier?

File: packages/mds-agency/api.ts [hunk 4]
```diff
@@ -59,15 +59,14 @@ function api(app: express.Express): express.Express {
...
-          // logger.info(providerName(provider_id), req.method, req.originalUrl)
         } else {
           return res.status(401).send({ error: 'authentication_error', error_description: 'Unauthorized' })
         }
       }
     } catch (error) {
       /* istanbul ignore next */
-      logger.error('request validation fail:', { originalUrl: req.originalUrl, error })
+      AgencyLogger.error('request claims parsing fail:', { originalUrl: req.originalUrl, error })
+      res.status(500).send(AgencyServerError)
```
Comment (drtyh2o) [inline packages/mds-agency/api.ts:69]: Normally, we add a `return` in front of `res.status` calls.

File: packages/mds-agency/request-handlers.ts [hunk 5]
```diff
@@ -180,7 +179,11 @@ export async function updateVehicleFail(
   } else if (!provider_id) {
     res.status(404).send({})
   } else {
-    logger.error(`fail PUT /vehicles/${device_id}`, { providerName: providerName(provider_id), body: req.body, error })
+    AgencyLogger.error(`fail PUT /vehicles/${device_id}`, {
+      providerName: providerName(provider_id),
```
Comment (drtyh2o) [inline packages/mds-agency/request-handlers.ts:183]: We should stop using the `providerName` function so we can eventually get rid of `mds-providers` altogether in favor of an async data source.

File: packages/mds-agency/utils.ts [hunk 6]
```diff
@@ -360,7 +359,7 @@ export async function readPayload(device_id: UUID): Promise<VehiclePayload> {
   try {
     payload.device = await db.readDevice(device_id)
   } catch (err) {
-    logger.error(err)
+    AgencyLogger.error(err)
```
Comment (drtyh2o) [inline packages/mds-agency/utils.ts:362]: Seems like just logging out the error without accompany text is less than ideal.

File: packages/mds-api-server/middleware/request-logging.ts [hunk 7]
```diff
@@ -65,11 +65,9 @@ export const RequestLoggingMiddleware = ({
     return next()
   },
   (req: ApiRequest, res: ApiResponse, next: express.NextFunction) => {
-    const { REQUEST_DEBUG } = process.env
-
-    if (REQUEST_DEBUG === 'true' && !excludePaths.some(path => req.path.match(path))) {
+    if (!excludePaths.some(path => req.path.match(path))) {
       const { path, params, query, body } = req
-      logger.info('REQUEST_DEBUG', { path, params, query, body })
+      ApiServerLogger.debug('Request details', { path, params, query, body })
```
Comment (drtyh2o) [inline packages/mds-api-server/middleware/request-logging.ts:70]: I've observed that `params` hasn't been parsed yet at this point in the middleware pipeline. It seems to always be `{}` so it doesn't have much value. The data will be in the path though, so perhaps we should just remove.

File: packages/mds-compliance-api/logger.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import logger from '@mds-core/mds-logger'
+
+export const ComplianceApiLogger = logger.createLogger('ComplianceApiLogger')
```
Comment (drtyh2o) [inline packages/mds-compliance-api/logger.ts:3]: I don't think you need `Logger` in the namespace name. Again, I am a fan of using the package name as the namespace to associate the logs with their corresponding modules.

**Review by drtyh2o — APPROVED** [review]
Some comments/nits, but good start. Thanks for doing this. I know it's tedious, but it will be super helpful.

File: packages/mds-agency-cache/logger.ts (hunk 3 above)
Comment (avatarneil) [reply packages/mds-agency-cache/logger.ts:3]: I thought about copy-pastaing the package name, but it feels a little redundant considering everything here has the `mds` prefix already added inherently by `mds-logger`

File: packages/mds-agency-cache/index.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-agency-cache/index.ts]: Fixed in a7334b7

File: packages/mds-agency-cache/index.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-agency-cache/index.ts]: Fixed in a7334b7

File: packages/mds-agency-cache/index.ts [hunk 9]
```diff
@@ -181,16 +180,15 @@ async function wipeDevice(device_id: UUID) {
     decorateKey(`device:${device_id}:device`)
   ]
   if (keys.length > 0) {
-    logger.info('mds-agency-cache::wipeDevice, deleting keys', { keys })
+    AgencyCacheLogger.debug('wipeDevice, deleting keys', { keys })
     return client.del(...keys)
   }
-  logger.warn('mds-agency-cache::wipeDevice, no keys found!', { device_id })
+  AgencyCacheLogger.warn('mds-agency-cache::wipeDevice, no keys found!', { device_id })
```
Comment (drtyh2o) [inline packages/mds-agency-cache/index.ts:186]: Unnecessary `mds-agency-cache` prefix on log text.

File: packages/mds-agency-cache/index.ts [hunk 10]
```diff
@@ -331,7 +327,9 @@ async function readDevicesStatus(query: {
   const deviceIds = deviceIdsRes.slice(skip, skip + take)
   const geoFinish = now()
   const timeElapsed = geoFinish - geoStart
-  logger.info(`mds-agency-cache readDevicesStatus bbox fetch ${JSON.stringify(bbox)} time elapsed: ${timeElapsed}ms`)
+  AgencyCacheLogger.debug(
+    `mds-agency-cache readDevicesStatus bbox fetch ${JSON.stringify(bbox)} time elapsed: ${timeElapsed}ms`
```
Comment (drtyh2o) [inline packages/mds-agency-cache/index.ts:331]: Unnecessary prefix.

File: packages/mds-agency-cache/index.ts [hunk 11]
```diff
@@ -352,7 +350,7 @@ async function readDevicesStatus(query: {
     .filter(item => Boolean(item))
   const eventsFinish = now()
   const eventsTimeElapsed = eventsFinish - eventsStart
-  logger.info(
+  AgencyCacheLogger.debug(
     `mds-agency-cache readDevicesStatus bbox check ${JSON.stringify(bbox)} time elapsed: ${eventsTimeElapsed}ms`
```
Comment (drtyh2o) [inline packages/mds-agency-cache/index.ts:354]: Maybe find all `mds-agency-cache`?

File: packages/mds-agency-cache/logger.ts (hunk 3 above)
Comment (drtyh2o) [reply packages/mds-agency-cache/logger.ts:3]: I don't think it's redundant. `DEBUG=mds` is not the same as `DEBUG=mds-agency-cache` and if we use the package name by convention then I can turn on debug logs without having to refer to the code to find out what namespace is being used.

File: packages/mds-geography/logger.ts [hunk 12]
```diff
@@ -0,0 +1,3 @@
+import logger from '@mds-core/mds-logger'
+
+export const GeographyLogger = logger.createLogger('mds-geography')
```
Comment (drtyh2o) [inline packages/mds-geography/logger.ts:3]: I wish we could rename some of these packages to include `-api` where appropriate. This really should be GeographyApiLogger, IMO.

File: packages/mds-agency-cache/index.ts [hunk 13]
```diff
@@ -245,7 +243,7 @@ async function readAllEvents(): Promise<Array<VehicleEvent | null>> {
   const keys = await readKeys('device:*:event')
   let finish = now()
   let timeElapsed = finish - start
-  logger.info(`MDS-DAILY /admin/events -> cache.readAllEvents() readKeys() time elapsed: ${timeElapsed}ms`)
+  AgencyCacheLogger.debug(`MDS-DAILY /admin/events -> cache.readAllEvents() readKeys() time elapsed: ${timeElapsed}ms`)
```
[the remark this answers is not in the document]
Comment (avatarneil) [reply packages/mds-agency-cache/index.ts:268]: 'tis not! I'll see if I can nuke this method.

**Pull request merged**